src/main/java/com/shapesecurity/salvation/directives/Directive.java

```diff
@@ -82,8 +82,12 @@ static void register(Class<? extends Directive> directiveClass) {
     @Nonnull private static <T> Set<T> union(@Nonnull Set<T> a, @Nonnull Set<T> b) {
         Set<T> set = new LinkedHashSet<>();
 
-        set.addAll(a);
-        set.addAll(b);
+        if (a.stream().allMatch(x -> x == None.INSTANCE) && a.equals(b)) {
+            return set;
+        }
+
+        set.addAll(a.stream().filter(x -> x != None.INSTANCE).collect(Collectors.toCollection(LinkedHashSet::new)));
```

Comment:
Just add them all unconditionally and then remove `None.INSTANCE` after.

Reply:
Lol indeed.

Reply:
Done.